src/task.c

```diff
@@ -372,20 +372,29 @@ static void ctx_switch(jl_ptls_t ptls, jl_task_t **pt)
                 " push %%ebp;\n" // instead of ESP
                 " jmp %P1;\n" // call `start_task` with fake stack frame
                 " ud2"
-                : : "r" (stackbase), "X"(&start_task) : "memory" );
+                : : "r"(stackbase), "X"(&start_task) : "memory" );
 #elif defined(_CPU_AARCH64_)
             asm(" mov sp, %0;\n"
                 " mov x29, xzr;\n" // Clear link register (x29) and frame pointer
                 " mov x30, xzr;\n" // (x30) to terminate unwinder.
-                " br %1;\n" // call `start_task` with fake stack frame
+                " b %1;\n" // call `start_task` with fake stack frame
                 " brk #0x1" // abort
-                : : "r" (stackbase), "r"(&start_task) : "memory" );
+                : : "r"(stackbase), "S"(&start_task) : "memory" );
+#elif defined(_CPU_ARM_)
+            // A "i" constraint on `&start_task` works only on clang and not on GCC.
+            asm(" mov sp, %0;\n"
+                " mov lr, #0;\n" // Clear link register (lr) and frame pointer
+                " mov fp, #0;\n" // (fp) to terminate unwinder.
+                " b start_task;\n" // call `start_task` with fake stack frame
```

Comment:
This assumes that we'll emit `.start_task` as an asm symbol (despite trying to tell the compiler not to, with use of `static` and such). What was wrong with using indirect branch?

Reply:
Or add `lr` and `fp` to the clobbers list

Reply:
> despite trying to tell the compiler not to

No we don't. That's what the `JL_USED_FUNC` is for. We are explicitly telling the compiler to emit that symbol.

> What was wrong with using indirect branch?

More instructions, but more importantly, I can't find a better way to make sure the compiler don't use, `sp`, `lr` or `fp` as the register to pass in the address. It seems that `lr` is most likely a valid choise if the compiler want to do that.

Reply:
That's what the clobbers list is for, although the compiler may also potentially attempt to save/restore those registers around it. Making the first instruction `mov r1, %1` is another way to ensure it, although in that case, using `{r1}` as the constraint is probably easier.

Reply:
> using {r1} as the constraint is probably easier.

That's not supported.

> That's what the clobbers list is for,

I want to be robust against different compiler options (they might have special behavior on `fp`). GCC errors when `fp` is in the clobber list but not for `sp` and `lr`.

> Making the first instruction `mov r1, %1` is another way to ensure it,

Thought about that, which is why I decided that just using a branch directly is simpler. The `used` attribute is explicitly documented for use like this.